datadog_lambda/wrapper.py

```diff
@@ -103,33 +107,42 @@ def __init__(self, func):
     def __call__(self, event, context, **kwargs):
         """Executes when the wrapped function gets called"""
         self._before(event, context)
+        response = None
         try:
-            return self.func(event, context, **kwargs)
+            response = self.func(event, context, **kwargs)
+            return response
         except Exception:
             submit_errors_metric(context)
             if self.span:
                 self.span.set_traceback()
             raise
         finally:
-            self._after(event, context)
+            self._after(event, context, response)
 
     def _before(self, event, context):
         try:
 
             set_cold_start()
             submit_invocations_metric(context)
-            # Extract Datadog trace context from incoming requests
-            dd_context = extract_dd_trace_context(event)
+            # Extract trigger tags and add them to a subsegment if found
+            trigger_tags = extract_trigger_tags(event, context)
+            if trigger_tags:
+                create_dd_root_span_metadata_subsegment(trigger_tags)
```

Comment:
BTW, just a reminder, i recall you would have to duplicate the root span tags to the `aws.lambda.function` span as well, right?

Reply:
Correct 👍